src/Storages/ObjectStorage/DataLakes/Iceberg/PartitionPruning.cpp

```diff
@@ -0,0 +1,138 @@
+    {
+        return Iceberg::PartitionTransform::Identity;
+    }
+    else if (transform_name == "void")
+    {
+        return Iceberg::PartitionTransform::Void;
+    }
+    else
+    {
+        return Iceberg::PartitionTransform::Unsupported;
+    }
+}
+
+// This function is used to convert the value to the start of the corresponding time period (internal ClickHouse representation)
+DateLUTImpl::Values getDateLUTImplValues(Int32 value, Iceberg::PartitionTransform transform)
+{
+    switch (transform)
+    {
+        case Iceberg::PartitionTransform::Year:
+            return DateLUT::instance().lutIndexByYearSinceEpochStartsZeroIndexing(value);
+        case Iceberg::PartitionTransform::Month:
+            return DateLUT::instance().lutIndexByMonthSinceEpochStartsZeroIndexing(static_cast<UInt32>(value));
+        case Iceberg::PartitionTransform::Day:
+            return DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value));
+        case Iceberg::PartitionTransform::Hour:
+        {
+            DateLUTImpl::Values values = DateLUT::instance().getValues(static_cast<ExtendedDayNum>(value / 24));
+            values.date += (value % 24) * 3600;
+            return values;
+        }
+        default:
+            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition transform for get values function: {}", transform);
+    }
+}
+
+// This function is used to convert the value to the start of the corresponding time period (in seconds)
+Int64 getTime(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Values values = getDateLUTImplValues(value, transform);
+    return values.date;
+}
+
+// This function is used to convert the value to the start of the corresponding date period (in days)
+Int16 getDay(Int32 value, Iceberg::PartitionTransform transform)
+{
+    DateLUTImpl::Time got_time = getTime(value, transform);
+    return DateLUT::instance().toDayNum(got_time);
+}
+
+
+Range getPartitionRange(
+    Iceberg::PartitionTransform partition_transform, size_t index, ColumnPtr partition_column, DataTypePtr column_data_type)
+{
+    if (partition_transform == Iceberg::PartitionTransform::Unsupported)
+    {
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition transform: {}", partition_transform);
+    }
+    if (partition_transform == Iceberg::PartitionTransform::Identity)
+    {
+        Field entry = (*partition_column.get())[index];
+        return Range{entry, true, entry, true};
+    }
+    if (partition_column->getDataType() != TypeIndex::Int32)
+    {
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Unsupported partition column type: {}", partition_column->getFamilyName());
+    }
+
+    LOG_DEBUG(&Poco::Logger::get("PartitionPruning"), "Column data type: {}", column_data_type->getName());
+
+    auto nested_data_type = column_data_type;
+    while (nested_data_type->isNullable())
+        nested_data_type = removeNullable(nested_data_type);
```

Comment:
It's not possible to have multiple nested `Nullable`. Also `removeNullable` returns the type iteslf if it's not actually `Nullable`

```suggestion
    auto nested_data_type = removeNullable(nested_data_type);
```

Reply:
Ok, I will replace while with if

Reply:
`if` is not needed, because `removeNullable` already has this if:
```c++
DataTypePtr removeNullable(const DataTypePtr & type)
{
    if (type->isNullable())
        return static_cast<const DataTypeNullable &>(*type).getNestedType();
    return type;
```